homeassistant/components/overkiz/climate_entities/atlantic_pass_apc_zone_control_zone.py

```diff
@@ -0,0 +1,249 @@
+            )
+        else:
+            await self.executor.async_execute_command(
+                OverkizCommand.SET_COOLING_ON_OFF,
+                OverkizCommandParam.ON,
+            )
+            await self.executor.async_execute_command(
+                OverkizCommand.SET_HEATING_ON_OFF,
+                OverkizCommandParam.ON,
+            )
+
+        await self.async_refresh_modes()
+
+    @property
+    def preset_mode(self) -> str:
+        """Return the current preset mode, e.g., schedule, manual."""
+
+        if self.is_using_derogated_temperature_fallback:
+            return super().preset_mode
+
+        mode = OVERKIZ_MODE_TO_PRESET_MODES[
+            cast(
+                str,
+                self.executor.select_state(
+                    OverkizState.IO_PASS_APC_HEATING_MODE
+                    if self.zone_control_hvac_mode == HVACMode.COOL
+                    else OverkizState.IO_PASS_APC_COOLING_MODE
+                ),
+            )
+        ]
+
+        return mode if mode is not None else PRESET_NONE
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+
+        if self.is_using_derogated_temperature_fallback:
+            return await super().async_set_preset_mode(preset_mode)
+
+        mode = PRESET_MODES_TO_OVERKIZ[preset_mode]
+
+        # For consistency, it is better both are synced like on the Thermostat.
+        await self.executor.async_execute_command(
+            OverkizCommand.SET_PASS_APC_HEATING_MODE, mode
+        )
+        await self.executor.async_execute_command(
+            OverkizCommand.SET_PASS_APC_COOLING_MODE, mode
+        )
+
+        await self.async_refresh_modes()
+
+    @property
+    def target_temperature(self) -> float:
+        """Return hvac target temperature."""
+
+        if self.is_using_derogated_temperature_fallback:
+            return super().target_temperature
+
+        if self.zone_control_hvac_mode == HVACMode.COOL:
+            return cast(
+                float,
+                self.executor.select_state(
+                    OverkizState.CORE_COOLING_TARGET_TEMPERATURE
+                ),
+            )
+
+        if self.zone_control_hvac_mode == HVACMode.HEAT:
+            return cast(
+                float,
+                self.executor.select_state(
+                    OverkizState.CORE_HEATING_TARGET_TEMPERATURE
+                ),
+            )
+
+        # Not sure this temperature is relevant.
+        return cast(
+            float, self.executor.select_state(OverkizState.CORE_TARGET_TEMPERATURE)
+        )
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new temperature."""
+
+        if self.is_using_derogated_temperature_fallback:
+            return await super().async_set_temperature(**kwargs)
+
+        temperature = kwargs[ATTR_TEMPERATURE]
+
+        # Change both (heating/cooling) temperature is a good way to have consistency
```

Comment:
Can you elaborate on this? Don't you want to set them separately based on the current mode?

Reply:
I am not an expert on climate devices and the implementations, but HA also offers `target_temp_high` and `target_temp_low` for devices in `heat_cool` mode. Here you can set 2 values.

https://www.home-assistant.io/integrations/climate/#service-climateset_temperature

Reply:
Let's see if this is required in a follow up PR.